### ISSUE TITLE
Non reporting errors for redundant space between "typeof" and "(" - related to "space-unary-ops" rule.

### ISSUE THREAD

**Issue by tomas-sentkeresty** [eslint/eslint#9907]
**Tell us about your environment**

* **ESLint Version:** v4.16.0
* **Node Version:** v6.10.2
* **npm Version:** v5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```json
{
    "extends": "airbnb-base",
    "rules": {
        "prefer-arrow-callback": "off",
        "func-names": ["error", "never"],
        "space-before-function-paren": ["error", "never"],
        "indent": ["error", 4],
        "no-var": "off",
        "object-shorthand": ["error", "never"],
        "prefer-destructuring": "off",
        "import/no-unresolved": [2, {"commonjs": true}],
        "import/no-dynamic-require": "warn",
        "global-require": "warn",
        "vars-on-top": "off",
        "no-param-reassign": "off",
        "eqeqeq": "off",
        "no-use-before-define": ["error", {"functions": false}],
        "prefer-template": "off",
        "no-shadow": "off",
        "max-len": "off",
        "comma-dangle": ["error", "never"],
        "no-restricted-syntax": ["error", "BinaryExpression[operator='in']"],
        "no-continue": "off",
        "no-plusplus": "off",
        "brace-style": ["error", "stroustrup", {"allowSingleLine": false}],
        "space-unary-ops": [2, {
            "overrides": {
                "typeof": false
            }
        }]
    }
}
```
> Last config is critical
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var obj = null;
if (typeof (obj) === 'object') {
    console.log("THIS SHOULD REPORT ERROR BETWEEN 'typeof' AND '('. BUT IT DOESN'T");
}
if (typeof(obj) === 'object') {
    console.log('THIS IS OK AS IT DOES NOT REPORT ERROR.');
}
```

```bash
./node_modules/.bin/eslint issue.js
```

**What did you expect to happen?**
First case with `typeof (obj) === 'object'` should report error.

**What actually happened? Please include the actual, raw output from ESLint.**
No error was reported in first case. Actually there is nothing to show from console.

**Comment by tomas-sentkeresty** [comment]
Closing in respect to #9906.

**Comment by platinumazure** [comment]
Reopening so I can tie this to my pull request.

**Comment by bladedancer** [comment]
This fix breaks: `yield [1, 2, 3]`

Is that the expected behaviour?

**Comment by platinumazure** [comment]
Could you supply your config?

`yield[1,2,3]` is valid JavaScript and so `yield [1,2,3]` should be flagged
if using "words: false" setting. The rule was already working for object
expressions (`yield {foo: 1}`). So we fixed the rule so it would apply for
all non-word tokens after `yield`, basically.
